src/WsSecurity.php

```diff
@@ -8,35 +8,50 @@ class WsSecurity
      * @var Security
      */
     protected $security;
+
     /**
      * @param string $username
      * @param string $password
-     * @param bool $passwordDigest
-     * @param int $addCreated
-     * @param int $addExpires
-     * @param bool $mustunderstand
+     * @param bool   $passwordDigest
+     * @param int    $addCreated
+     * @param int    $addExpires
+     * @param bool   $mustunderstand
      * @param string $actor
      * @param string $usernameId
-     * @param bool $addNonce
+     * @param bool   $addNonce
+     * @param string $envelopeNamespace
      */
-    protected function __construct($username, $password, $passwordDigest = false, $addCreated = 0, $addExpires = 0, $mustunderstand = false, $actor = null, $usernameId = null, $addNonce = true)
-    {
+    protected function __construct(
+        $username,
+        $password,
+        $passwordDigest = false,
+        $addCreated = 0,
+        $addExpires = 0,
+        $mustunderstand = false,
+        $actor = null,
+        $usernameId = null,
+        $addNonce = true,
+        $envelopeNamespace = 'SOAP-ENV'
+    ) {
         $this
-            ->initSecurity($mustunderstand, $actor)
+            ->initSecurity($mustunderstand, $actor, $envelopeNamespace)
             ->setUsernameToken($username, $usernameId)
             ->setPassword($password, $passwordDigest, $addCreated)
             ->setNonce($addNonce)
             ->setCreated($addCreated)
             ->setTimestamp($addCreated, $addExpires);
     }
+
     /**
-     * @param bool $mustunderstand
+     * @param bool   $mustunderstand
      * @param string $actor
+     * @param string $envelopeNamespace
+     *
      * @return WsSecurity
      */
-    protected function initSecurity($mustunderstand = false, $actor = null)
+    protected function initSecurity($mustunderstand = false, $actor = null, $envelopeNamespace)
```

Comment:
In my opinion, having a method with the two first parameters as optional should not have a third required parameter that would require to set the two first parameters anytime.
```suggestion
    protected function initSecurity($mustunderstand = false, $actor = null, $envelopeNamespace = Security::ENV_NAMESPACE)
```